sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -1127,6 +1145,54 @@ public enum JsonPropertyNames
         UseExact = 0,
         CamelCase = 1,
     }
+    public partial class ModelJsonConverter : System.Text.Json.Serialization.JsonConverter<Azure.Core.Serialization.IModelJsonSerializable<object>>
+    {
+        public ModelJsonConverter() { }
+        public ModelJsonConverter(Azure.Core.Serialization.ModelSerializerFormat format) { }
+        public ModelJsonConverter(Azure.Core.Serialization.ModelSerializerOptions options) { }
+        public Azure.Core.Serialization.ModelSerializerOptions ModelSerializerOptions { get { throw null; } }
+        public override bool CanConvert(System.Type typeToConvert) { throw null; }
+        public override Azure.Core.Serialization.IModelJsonSerializable<object> Read(ref System.Text.Json.Utf8JsonReader reader, System.Type typeToConvert, System.Text.Json.JsonSerializerOptions options) { throw null; }
+        public override void Write(System.Text.Json.Utf8JsonWriter writer, Azure.Core.Serialization.IModelJsonSerializable<object> value, System.Text.Json.JsonSerializerOptions options) { }
+    }
+    public static partial class ModelSerializer
+    {
+        public static object Deserialize(System.BinaryData data, System.Type returnType, Azure.Core.Serialization.ModelSerializerFormat format) { throw null; }
+        public static object Deserialize(System.BinaryData data, System.Type returnType, Azure.Core.Serialization.ModelSerializerOptions? options = null) { throw null; }
+        public static T Deserialize<T>(System.BinaryData data, Azure.Core.Serialization.ModelSerializerFormat format) where T : Azure.Core.Serialization.IModelSerializable<T> { throw null; }
+        public static T Deserialize<T>(System.BinaryData data, Azure.Core.Serialization.ModelSerializerOptions? options = null) where T : Azure.Core.Serialization.IModelSerializable<T> { throw null; }
+        public static System.BinaryData Serialize(object model, Azure.Core.Serialization.ModelSerializerFormat format) { throw null; }
+        public static System.BinaryData Serialize(object model, Azure.Core.Serialization.ModelSerializerOptions? options = null) { throw null; }
+        public static System.BinaryData SerializeCore(Azure.Core.Serialization.IModelJsonSerializable<object> model, Azure.Core.Serialization.ModelSerializerOptions options) { throw null; }
```

Comment:
This was to replace the ModelWriter that we made internal.  We wanted a "scary name" since its not really intended for customers since you can get into odd situations like passing in wire to this method and the wire format is actually xml.

Reply:
Cool, sounds good.  Would it make sense to indicate in the doc comment that it is only intended to be called from within Models to serialize themselves?  Another pattern to consider is making it a separate type so separate APIs for end-user customers and library authors, but I don't feel strongly about that if the doc comment is clear.

Reply:
Yeah this was all brought up in the meeting last night and the decision was to go with this name on this class.  It used to be on a class called ModelWriter before it moved here.